samples/hello_world/src/main.c

```diff
@@ -8,6 +8,13 @@
 
 int main(void)
 {
-	printf("Hello World! %s\n", CONFIG_BOARD);
+	printf("Hello World! %s\n",
+#ifdef CONFIG_BOARD_TARGET
+	       CONFIG_BOARD_TARGET
+#else
+	       CONFIG_BOARD
+#endif
+	);
```

Comment:
```suggestion
	printf("Hello World! %s\n", CONFIG_BOARD_TARGET);
```

let's not make hello world uglier.

Reply:
Not possible because it doesn't exist for hwmv1 devices, which are still supported

Reply:
then I'd define `CONFIG_BOARD_TARGET` to the equivalent when in HWMv1

Reply:
No, we should not be adding hwmv2 defines to hwmv1 boards

Reply:
then we need a solution

Reply:
Well I mean it doesn't considering the term was coined in the decision that concluded on Friday, it was simple a "board" in hwmv1

Reply:
then why not insert BOARD_TARGET = BOARD in v1 Kconfig? this will make life easier in all samples relying on BOARD_TARGET.

Reply:
Because it's a hwmv2 concept not a hwmv1 concept and can be used from application code to detect a hwmv2 board

Reply:
> can be used from application code to detect a hwmv2 board

where is this needed?

Reply:
Anywhere?

Reply:
It's not passed in C to the application which is where this is

Reply:
> `CONFIG_BOARD_TARGET` should always be defined.
> 
> Although we have discussed terminology in context of HWMv2, then I would argue that for HWMv1 we should have `CONFIG_BOARD_TARGET` == `CONFIG_BOARD` simply because HWMv1 doesn't have the concept of qualifiers.

I do not agree, hwmv1 is now end of life, it doesn't make sense to introduce hwmv2 with terms **specifically for hwmv2** then just apply them to hwmv1 for releases 3.7 and 4.0 when it's not in any of the previous released when we all agreed that hwmv2 is now the standard

Reply:
we simply need `CONFIG_BOARD_TARGET` defined everywhere, forcing people to use ifdeffery just for an obsolete hw model doesn't make any sense.

Reply:
> we simply need `CONFIG_BOARD_TARGET` defined everywhere, forcing people to use ifdeffery just for an obsolete hw model doesn't make any sense.

Adding features to a deprecated and unsupported hardware model where the terms were never applied doesn't make any sense

Reply:
> Adding features to a deprecated and unsupported hardware model where the terms were never applied doesn't make any sense

This is about being pragmatic.
Comparing the two options of:
Adding `CONFIG_BOARD_TARGET` one place vs having `#ifdef CONFIG_BOARD_TARGET` n-places.

then I choose the first, "Adding `CONFIG_BOARD_TARGET` one place".

Reply:
>  Adding CONFIG_BOARD_TARGET one place vs having #ifdef CONFIG_BOARD_TARGET n-places.

We're not adding it to multiple places, we're adding it to hello world, and that's it

Reply:
> > Adding CONFIG_BOARD_TARGET one place vs having #ifdef CONFIG_BOARD_TARGET n-places.
> 
> We're not adding it to multiple places, we're adding it to hello world, and that's it

Other samples (eg multitarget hello world, mbox, IPC) which typically run on multiple cores will also need this to become more user-friendly. Spreading such ifdeffery everywhere for no added value doesn't seem like a good idea.

Reply:
> We're not adding it to multiple places, we're adding it to hello world, and that's it

No, any place which would want to print `CONFIG_BOARD_TARGET` would need to do this.
So might be one place in this particular PR, but would end up in n-places later.

Reply:
it could be in cases like this where knowing the CPU cluster on a multicore SoC would be nice:
https://github.com/zephyrproject-rtos/zephyr/blob/42c8b7fb00a8f6343966e140b4bba593bb92cfdb/tests/boot/test_mcuboot/src/main.c#L14

or when printing board in `userspace/hello_world`, then why should that printing deviate from simple `hello_world` ?
https://github.com/zephyrproject-rtos/zephyr/blob/42c8b7fb00a8f6343966e140b4bba593bb92cfdb/samples/userspace/hello_world_user/src/main.c#L20-L22